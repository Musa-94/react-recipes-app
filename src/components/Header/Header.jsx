import React from 'react';

function Header({ children, title, styleClass }) {
    return (
        <header>
            <div className='container-fluid'>
                <div className={`row align-items-center ${styleClass}`}>
                    <div className="col text-center text-slanted">
                        <h1 className={'text-light text-uppercase display-3 letter-spacing text-slander'}>
                            {title}
                        </h1>
                        {children}
                    </div>
                </div>
            </div>
        </header>
    );
}

Header.defaultProps = {
    title: 'default title',
    styleClass: 'header-hero'
}

export default React.memo(Header);
