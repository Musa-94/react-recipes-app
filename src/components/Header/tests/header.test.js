import React from 'react';
import Header from '../Header';

const props = {
    title: 'title',
    children: 'children',
    styleClass: 'styleClass',
};

describe('Header snapshot', () => {
    it('should correct renderer', () => {
        const component = renderer(<Header {...props}/>);

        expect(component).toMatchSnapshot();
    });
});