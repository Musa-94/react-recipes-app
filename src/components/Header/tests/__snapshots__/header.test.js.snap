// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot should correct renderer 1`] = `
<header>
  <div
    className="container-fluid"
  >
    <div
      className="row align-items-center styleClass"
    >
      <div
        className="col text-center text-slanted"
      >
        <h1
          className="text-light text-uppercase display-3 letter-spacing text-slander"
        >
          title
        </h1>
        children
      </div>
    </div>
  </div>
</header>
`;
